src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -96,8 +109,29 @@ public Task Start(IMessageSession session)
             }
         }
 
+        CancellationTokenSource LongRunningWarning(string message)
+        {
+            var tokenSource = new CancellationTokenSource();
+
+            // ReSharper disable once MethodSupportsCancellation
+            Task.Run(async () =>
+            {
+                try
+                {
+                    await Task.Delay(LongRunningWarningTimeSpan, tokenSource.Token);
+                    Log.Warn(message);
+                }
+                catch (TaskCanceledException)
+                {
+                }
+            });
+
+            return tokenSource;
+        }
+
         IEnumerable<IWantToRunWhenEndpointStartsAndStops> wantToRunWhenBusStartsAndStops;
         ConcurrentBag<IWantToRunWhenEndpointStartsAndStops> thingsRanAtStartup = new ConcurrentBag<IWantToRunWhenEndpointStartsAndStops>();
-        static ILog Log = LogManager.GetLogger<StartableAndStoppableRunner>();
+        public static ILog Log = LogManager.GetLogger<StartableAndStoppableRunner>();
+        public TimeSpan LongRunningWarningTimeSpan { get; set; }
```

Comment:
I'm not a huge fan either but I feel like the ROI on these changes is low. What is wrong with Moq in the tests project?